### DISCUSSION TITLE
A few observation or bugs in Cooja 4.9

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2637] [Q&A] [answered]
I fetched the new Contiki 4.9 yesterday.

Gradle 7.6 is the current version, it used to be 8.1. I'm not sure if this was Intentional.

The following can be seen in the attached image if needed.

I have no idea where did the `FATAL [sim] (AbstractRadioMedium.java:195) - Connection source turned off radio: Radio at Contiki 1` message comes from. I did not turn off any radio. I started Cooja, uploaded an existing simulation, and ran it. The `FATAL` however sounds ominous, and it occurred multiple times. What should I fix?

What do the following mean?:
- WARNING: Using incubator modules: jdk.incubator.foreign
- WARN [SwingWorker-pool-2-thread-1] (Cooja.java:1564) - Loaded simulation may depend on not found extension: '/home/user/contiki-ng/tools/cooja/[APPS_DIR]/avrora'

Lastly, tell Cooja to Append its output to a log file and run it. Then exit Cooja. When Cooja is invoked again, it will remember it was appending in its prior run (the check-mark on Append in the menu will still be there). I think that this is the cause for the WARN `[AWT-EventQueue-0]...` in the attached figure.

Thank you for thinking about these. Sincerely,
Rahav

![Screenshot from 2023-08-22 18-42-04](https://github.com/contiki-ng/contiki-ng/assets/2912950/e6ebf4ae-ae2f-45a9-9fda-9929474e1462)

**Comment by rahav** [comment]
The warning on `'/home/user/contiki-ng/tools/cooja/[APPS_DIR]/avrora'` appears each time a new simulation is run.

**Comment by pjonsson** [comment, marked as answer]
There was an accidental downgrade of Cooja on develop. It's been fixed, so please update again and you should be running Gradle 8.3.

The incubator thing is Cooja-internal and nothing to worry about (Java doesn't allow us to suppress the warning).

The avrora warning is something old from your simulation file. You can safely remove the avrora line from your simulation file.

If you want to do an automatic update of the simulation file (you will get a lot more changes), make a backup of the simulation file, and run `cooja --args='--update-simulation simulation.csc'`. The GUI will seem to start, and then Cooja will quit after updating the simulation file.

The radio message is probably from a race condition in the closing of the simulation, and if that is when it happens, it's harmless despite the dramatic FATAL label (if it's not the closing, something went wrong and the radio was lost, which is dramatic). Are you still getting the radio messages after updating to latest develop?

**Reply by rahav** [reply]
Thank you, Cooja output to the console looks great now. Running `cooja --args...` fixed the avrora problem.

`pull` of the latest `develop` branch brought Gradle 8.3 back. In the few sims that I ran, there were no FATAL radio messages.

**Comment by rahav** [comment]
@pjonsson Interesting enough to consider. The
````
FATAL [sim] (AbstractRadioMedium.java:195) - Connection source turned off radio: Radio at Contiki 1
````
message appears when clicking the `Reload` button. In other words, no simulation is running.

**Reply by pjonsson** [reply]
Ok, that is the shutdown race I mentioned. It's annoying but you don't need to worry about that one, it isn't fatal despite the message saying so.

**Reply by rahav** [reply]
THANK YOU!
